x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -302,9 +300,9 @@ static Set<String> fieldNames(LogicalPlan parsed, Set<String> enrichPolicyMatchF
                 references.addAll(enrichRefs);
             } else {
                 references.addAll(p.references());
-                // TODO: Is this still needed? This branch used to live in UnresolvedRelation.references().
-                if (p instanceof UnresolvedRelation ur && ur.indexMode() == IndexMode.TIME_SERIES) {
```

Comment:
index_mode time_series I think is more general and more appropriate for `metrics` command. I wouldn't change this to look at the aggregation type instead. My 2c (I am not very familiar with `metrics` but `index_mode == time_series` is something that's being discussed and it's a thing :-); see [this page from docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/tsds-index-settings.html)).

Reply:
Changed it back to rely on the index mode, as it was before (but left it inside fieldnames).